scanner/services/matcher.go

```diff
@@ -74,7 +74,15 @@ func (s *matcherService) GetVulnerabilities(ctx context.Context, req *v4.GetVuln
 		ir, err = getClairIndexReport(ctx, s.indexer, req.GetHashId())
 	} else {
 		zlog.Info(ctx).Msg("has contents, parsing")
-		ir, err = s.parseIndexReport(req.GetContents())
+		// Assume indexing was successful.
```

Comment:
Not sure if this comment helps. Is it possible that the Contents came from the caller from IndexReports that failed?

Reply:
> Assuming the only way people communicate with Scanner V4 is via our client library, then we should not be getting any requests for failed index reports.

This API was designed to allow matching over any "content manifest", so "assuming" the content came from a successful IR is not 100% accurate. I also noticed that the Matcher does not care about the IR successful state. So this piece of the logic is irrelevant.

All you need here to fix this is to return an IR with `Success=True` from the remote indexer.

Reply:
There is no harm in doing it, though, so let's move forward.